### PR TITLE
fix: custom attributes are ignored - #5084

### PR DESCRIPTION
**Pull request by zailic**
OK... i opened a new PR for #5084 issue, as i shot myself in a foot trying to modify a previous commit that I pushed under another github user.

### CODE REVIEW

File: instrumentation/net/http/otelhttp/labeler.go [hunk 1]
```diff
@@ -48,7 +48,10 @@ type labelerContextKeyType int
 
 const lablelerContextKey labelerContextKeyType = 0
 
-func injectLabeler(ctx context.Context, l *Labeler) context.Context {
+// InjectLabelerIntoContext helps to inject a Labeler into a context. Being a public
+// function, it is safe to use in any package that wants to add custom attributes to
+// the metrics recorded by the net/http instrumentation.
+func InjectLabelerIntoContext(ctx context.Context, l *Labeler) context.Context {
```
Comment (dmathieu) [inline instrumentation/net/http/otelhttp/labeler.go:54]: How about naming this methid `ContextWithLabeler`?
That would better match the similar method names we have in the trace API.
https://github.com/open-telemetry/opentelemetry-go/blob/main/trace/context.go

Comment (zailic) [reply instrumentation/net/http/otelhttp/labeler.go:54]: Yeah, you're right! I'll update the PR accordingly!

**Comment by zailic** [comment]
@dmathieu - Could you please let me know if there's any changes or  information from me in order to move forward with this PR?

File: instrumentation/net/http/otelhttp/labeler.go [hunk 2]
```diff
@@ -37,8 +37,12 @@ type labelerContextKeyType int
 
 const lablelerContextKey labelerContextKeyType = 0
 
-func injectLabeler(ctx context.Context, l *Labeler) context.Context {
-	return context.WithValue(ctx, lablelerContextKey, l)
+// ContextWithLabeler returns a new context with the provided Labeler instance.
+// Attributes added to the specified labeler will be injected into metrics
+// emitted by the instrumentation.Only one labeller can be injected into the
```
Comment (dmathieu) [inline instrumentation/net/http/otelhttp/labeler.go:42]:
```suggestion
// emitted by the instrumentation. Only one labeller can be injected into the
```

**Comment by zailic** [comment]
Hi @dmathieu , when you have a moment, could you please take a look at this PR? Thank you!

**Comment by dmathieu** [comment]
There is still a lint issue.

**Comment by zailic** [comment]
> There is still a lint issue.

@dmathieu  Merged the latest main branches and did a `make precommit` on my local...I hope this will solve the failing build step.

**Comment by dmathieu** [comment]
The issue appears to be that `go.sum` is changed by the linter. You may have to run `go mod tidy`

**Comment by zailic** [comment]
> The issue appears to be that `go.sum` is changed by the linter. You may have to run `go mod tidy`

it's strange i did a `make precommit`(which incluedes a `go mod tidy` step) before the push...

**Comment by zailic** [comment]
> The issue appears to be that `go.sum` is changed by the linter. You may have to run `go mod tidy`
@dmathieu - i pushed the modified `go.sum` -  could you please give it another try?

File: instrumentation/net/http/otelhttp/go.sum [hunk 3]
```diff
@@ -17,8 +17,14 @@ go.opentelemetry.io/otel v1.26.0 h1:LQwgL5s/1W7YiiRwxf03QGnWLb2HW4pLiAhaA5cZXBs=
 go.opentelemetry.io/otel v1.26.0/go.mod h1:UmLkJHUAidDval2EICqBMbnAd0/m2vmpf/dAM+fvFs4=
 go.opentelemetry.io/otel/metric v1.26.0 h1:7S39CLuY5Jgg9CrnA9HHiEjGMF/X2VHvoXGgSllRz30=
 go.opentelemetry.io/otel/metric v1.26.0/go.mod h1:SY+rHOI4cEawI9a7N1A4nIg/nTQXe1ccCNWYOJUrpX4=
+go.opentelemetry.io/otel/sdk v1.26.0 h1:Y7bumHf5tAiDlRYFmGqetNcLaVUZmh4iYfmGxtmz7F8=
+go.opentelemetry.io/otel/sdk v1.26.0/go.mod h1:0p8MXpqLeJ0pzcszQQN4F0S5FVjBLgypeGSngLsmirs=
+go.opentelemetry.io/otel/sdk/metric v1.26.0 h1:cWSks5tfriHPdWFnl+qpX3P681aAYqlZHcAyHw5aU9Y=
+go.opentelemetry.io/otel/sdk/metric v1.26.0/go.mod h1:ClMFFknnThJCksebJwz7KIyEDHO+nTB6gK8obLy8RyE=
```
[the remark this answers is not in the document]
Comment (zailic) [reply instrumentation/net/http/otelhttp/go.sum:23]: oh yes... good point! I'll do that. Thanks!

**Review by dmathieu — APPROVED** [review]

**Comment by dmathieu** [comment]
Something is wrong with your merges.

**Comment by zailic** [comment]
> Something is wrong with your merges.

strange... i only merged latest main branch changes without having any conflicts

**Comment by dmathieu** [comment]
PRs need at least 2 reviewers, and can only be merged by a maintainer. So yes, more reviews are required.
Also, while the diff of your PR looks good, there are a lot of commits attached to it, which cause a CLA issue. It won't be mergeable that way.

**Comment by zailic** [comment]
> PRs need at least 2 reviewers, and can only be merged by a maintainer. So yes, more reviews are required. Also, while the diff of your PR looks good, there are a lot of commits attached to it, which cause a CLA issue. It won't be mergeable that way.

Understood. Thanks! I've squashed bloating commits.

**Comment by zailic** [comment]
Hi @Aneurysm9,

I hope you're doing well. I would appreciate your review when you have a moment. 

Thanks!

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**